### PR TITLE
MODINREACH-120 Record Contribution: Data model for job execution status tracking and basic statistics

### PR DESCRIPTION
**Pull request by oliinyko**
## Purpose
Provide data model to support tracking of Initial Record Contribution process. Data model should support tracking for both, the latest and all previous runs.

JIRA: [MODINREACH-120](https://issues.folio.org/browse/MODINREACH-120)

## Approach
Create data model and repository

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/innreach/repository/ContributionRepository.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import static org.folio.innreach.domain.entity.Contribution.FETCH_HISTORY_COUNT_QUERY_NAME;
+
+@Repository
+public interface ContributionRepository extends JpaRepository<Contribution, UUID> {
+
+  @Query(name = Contribution.FETCH_CURRENT_QUERY_NAME)
+  Optional<Contribution> fetchCurrentByCsId(UUID id);
+
+  @Query(name = Contribution.FETCH_HISTORY_QUERY_NAME, countName = FETCH_HISTORY_COUNT_QUERY_NAME)
+  Page<Contribution> fetchHistoryByCsId(UUID id, Pageable pageable);
```
Comment (dmtkachenko) [inline src/main/java/org/folio/innreach/repository/ContributionRepository.java]: same here

File: src/test/java/org/folio/innreach/fixture/ContributionFixture.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    contribution.getErrors().forEach(e -> e.setContribution(contribution));
+
+    return contribution;
+  }
+
+  public static CentralServer refCentralServer() {
+    return TestUtil.refCentralServer(PRE_POPULATED_CENTRAL_SERVER_UUID);
+  }
+
+  public static AgencyLocationLscMapping findLocalServerMappingByCode(AgencyLocationMapping mapping, String code) {
```
Comment (maryna-steshenko) [inline src/test/java/org/folio/innreach/fixture/ContributionFixture.java]: Is this method relevant to this task?

**Review by dmtkachenko — APPROVED** [review]
